src/endpoints/user-status/domain/use-cases/get-user-status.use-case.ts

```diff
@@ -0,0 +1,102 @@
+import { inject, injectable, Result, UseCase } from '@alien-worlds/api-core';
+import { DacDirectory } from '@alien-worlds/eosdac-api-common';
+
+import {
+  DAOUserStatusType,
+  ERROR_MESSAGE_TYPE,
+} from '../../data/dtos/userstatus.dto';
+import { DaoUserStatus } from '../entities/dao-user-status';
+import { GetCandidateUseCase } from './get-candidate.use-case';
+import { GetCustodianUseCase } from './get-custodian.use-case';
+import { GetMemberTermsUseCase } from '../../../candidates/domain/use-cases/get-member-terms.use-case';
+import { GetMemberAgreedTermsUseCase } from './get-member-agreed-terms.use-case';
+
+/*imports*/
+/**
+ * @class
+ */
+@injectable()
+export class GetUserStatusUseCase implements UseCase<DaoUserStatus[]> {
+  public static Token = 'GET_USER_STATUS_USE_CASE';
+
+  constructor(
+    /*injections*/
+    @inject(GetCandidateUseCase.Token)
+    private getCandidatesUseCase: GetCandidateUseCase,
+    @inject(GetCustodianUseCase.Token)
+    private getCustodianUseCase: GetCustodianUseCase,
+    @inject(GetMemberTermsUseCase.Token)
+    private getMemberTermsUseCase: GetMemberTermsUseCase,
+    @inject(GetMemberAgreedTermsUseCase.Token)
+    private getMembersAgreedTermsUseCase: GetMemberAgreedTermsUseCase
+  ) {}
+
+  /**
+   * @async
+   * @returns {Promise<Result<DaoUserStatus[]>>}
+   */
+  public async execute(
+    walletId: string,
+    dacs: DacDirectory[]
+  ): Promise<Result<DaoUserStatus[]>> {
+    const result: DaoUserStatus[] = [];
+    let status: DAOUserStatusType = DAOUserStatusType.EXPLORER;
+    for (const dac of dacs) {
+      status = DAOUserStatusType.EXPLORER;
```

Comment:
May be we define this variable here, instead of Line 43?
Since, it is being assigned with every iteration so shouldn't affect.